plugins/tabletools/dialogs/tableCell.js

```diff
@@ -119,18 +119,49 @@ CKEDITOR.dialog.add( 'cellProperties', function( editor ) {
 			{
 				type: 'select',
 				id: 'cellType',
-				requiredContent: 'th',
+				requiredContent: 'th[scope]',
 				label: langCell.cellType,
 				'default': 'td',
 				items: [
 					[ langCell.data, 'td' ],
-					[ langCell.header, 'th' ]
+					[ langCell.columnHeader, 'thc' ],
+					[ langCell.rowHeader, 'thr' ]
 				],
 				setup: setupCells( function( selectedCell ) {
-					return selectedCell.getName();
+					var scope = selectedCell.getAttribute( 'scope' );
+
+					switch ( scope ) {
+						case 'row':
+							return 'thr';
+						case 'col':
+							return 'thc';
+						default:
```

Comment:
Before this change CKE4 accepted `th` without the `[scope]` attribute. In fact, it still accepts it (as otherwise it would be quite breaking change – some content would be lost after updating CKE4). However, it is not represented in the cell dialog – it's shown as a data cell, not a header.

Probably we would need to still support the old "Header" option alongside new ones or treat such `[scope]`-less `th` as one of the new ones by default.

Reply:
And this is also the reason the CI was dead. I'll correct that 👍